src/pages/Main/MainCalendarModal/MainCalendarModal.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { DateRange } from 'react-date-range';
+import styled from 'styled-components';
+import 'react-date-range/dist/styles.css';
+import 'react-date-range/dist/theme/default.css';
+
+const MainCalendarModal = ({ setModalVisible, setDate, date }) => {
+  const closeCalendarModal = e => {
+    e.target === e.currentTarget &&
+      setModalVisible(prev => ({ ...prev, calendarModal: false }));
+  };
+
+  return (
+    <ModalBackground onClick={closeCalendarModal}>
+      <ModaContainer>
+        <DateRange
+          minDate={new Date()}
+          months={2}
+          direction="horizontal"
+          editableDateInputs={true}
+          onChange={item => setDate([item.selection])}
+          moveRangeOnFirstSelection={false}
+          ranges={date}
+        />
+      </ModaContainer>
+    </ModalBackground>
+  );
+};
+
+export default MainCalendarModal;
+
+const ModalBackground = styled.div`
+  position: fixed;
+  display: flex;
+  top: 0;
+  justify-content: center;
+  align-items: center;
```

Comment:
연관된 속성끼리 붙여 써주면 더 좋을 것 같습니다.
```suggestion
  position: fixed;
  top: 0;
  display: flex;
  justify-content: center;
  align-items: center;
```